packages/slidev/node/export.ts

```diff
@@ -133,6 +148,18 @@ export async function exportSlides({
       printBackground: true,
       preferCSSPageSize: true,
     })
+
+    if (withTOC) {
+      const outlinePdf = outlinePdfFactory(pdfLib)
+      const pdf = fs.readFileSync(output, { encoding: 'base64' })
+      const outline = slides.map(({ title }, idx) =>
+        title ? `${idx + 1}||${title}` : null,
+      ).filter(outline => !!outline).join('\n')
```

Comment:
Don't you want to manage the "deepness" of title ?
In that case you may also be interested in the `level` property.

Also please heck the following properties (`rawTree`, `treeWithActiveStatuses` and `tree`) from here https://github.com/slidevjs/slidev/blob/main/packages/client/logic/nav.ts#L79-L86
I think It would better to use the same logic and thus the same functions than the ones used for the `<Toc>` component
And then having a dedicated renderer for the outline.

Reply:
Thank you for your comment! I did not know `<Toc />` component. I will refer it and change my logic to use same logic of `<Toc />`. However, I am wondering how to use `rawTree` in nodejs `slidev` because `rawTree` is plugged via a vite slidev plugin in client side build. 💭

Reply:
@tonai 

I'm sorry for the late response 🙇 

I changed to handle "deepness" of title in [18cf377](https://github.com/slidevjs/slidev/pull/669/commits/18cf377ea43a853503d8eb8be7a233fd829fabea).
![スクリーンショット 2022-08-14 17 52 37](https://user-images.githubusercontent.com/42742053/184529628-9f5c07d5-909c-4a35-9bb0-bd5b1205e333.png)

> Also please heck the following properties (rawTree, treeWithActiveStatuses and tree) from here https://github.com/slidevjs/slidev/blob/main/packages/client/logic/nav.ts#L79-L86
I think It would better to use the same logic and thus the same functions than the ones used for the <Toc> component
And then having a dedicated renderer for the outline.

I agree, but the current logic about `<Toc />` is too for client-side to share the logic also with `export` command in CLI…

To share the logic about `<Toc />` also with `export` command, I searched a package to share the logic in client-side and server-side, but I could not find that… Do you know sucha a package for this usecase?

Reply:
FYI: in the current implementation of this PR, the logic about ToC of PDF in `export` command is implemented adhocky separated from `<Toc />` component.

Reply:
@Shinyaigeek The idea of reuseing the same functions would be to ensure that the generated TOC will be the same as the one created with the `<Toc/>` component.
But yes those functions live in the client package.

@antfu what do you think of having a shared module that could be imported in either the backend or the frontend ?
Or should we duplicate some functions ? (here it is about all 3 `addToTree`, `getTreeWithActiveStatuses` and `filterTree` functions in `client/logic/nav.ts`)

Reply:
Yes, as @tonai says, it is better to share the logic which is reusable. If there is no package to share the utility with client-side and cli, perhaps we should make such a package before promoting this PR.